runtime/src/main/java/org/corfudb/runtime/exceptions/TrimmedException.java

```diff
@@ -1,15 +1,27 @@
 package org.corfudb.runtime.exceptions;
 
+import lombok.Getter;
+import lombok.Setter;
+
 /**
  * This exception is thrown when a client tries to read an address
  * that has been trimmed.
  */
 public class TrimmedException extends LogUnitException {
-    public TrimmedException() {
 
+    /*
+     * This flag determins whether the operation that caused this exception
+     * can retry or not.
+     */
+    @Getter
+    @Setter
+    private boolean retriable = true;
```

Comment:
I would consider renaming this "resolved" or something similar, a "resolved" trimmed exception means that we're certain this address is permanently trimmed and no longer accessible, a "unresolved" trimmed exception does not.